In [2]:
#import libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [3]:
#load movie data
movies = pd.read_csv("movies.csv")

In [4]:
movies.head()

poster  \
0  https://m.media-amazon.com/images/S/sash/4Fyxw...   
1  https://m.media-amazon.com/images/S/sash/4Fyxw...   
2  https://m.media-amazon.com/images/S/sash/4Fyxw...   
3  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                      title certificate  runtime                      genre  \
0  The Shawshank Redemption          15  142 min                      Drama   
1           The Dark Knight         12A  152 min       Action, Crime, Drama   
2                 Inception         12A  148 min  Action, Adventure, Sci-Fi   
3                Fight Club          18  139 min                      Drama   
4              Pulp Fiction          18  154 min               Crime, Drama   

   rating                                              about  \
0     9.3  Two imprisoned men bond over a number of years...   
1     9.0  When the menace known as the Joker wreaks havo...   
2     8.8  A thief who steals corporate secrets through t...   
3     8.8  An insomniac office worker and a devil-may-car...   
4     8.9  The lives of two mob hitmen, a boxer, a gangst...   

            director                                              stars  \
0     Frank Darabont  ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...   
1  Christopher Nolan  ('Christian Bale',), ('Heath Ledger',), ('Aaro...   
2  Christopher Nolan  ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...   
3      David Fincher  ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...   
4  Quentin Tarantino  ('John Travolta',), ('Uma Thurman',), ('Samuel...   

       votes gross_earn  
0  26,26,905    $28.34M  
1  25,98,173   $534.86M  
2  23,04,062   $292.58M  
3  20,71,088    $37.03M  
4  20,11,013   $107.93M

In [5]:
cleaned_movies = movies.copy()

In [6]:
cleaned_movies

poster  \
0     https://m.media-amazon.com/images/S/sash/4Fyxw...   
1     https://m.media-amazon.com/images/S/sash/4Fyxw...   
2     https://m.media-amazon.com/images/S/sash/4Fyxw...   
3     https://m.media-amazon.com/images/S/sash/4Fyxw...   
4     https://m.media-amazon.com/images/S/sash/4Fyxw...   
...                                                 ...   
4995  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4996  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4997  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4998  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4999  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                                title certificate         runtime  \
0            The Shawshank Redemption          15         142 min   
1                     The Dark Knight         12A         152 min   
2                           Inception         12A         148 min   
3                          Fight Club          18         139 min   
4                        Pulp Fiction          18         154 min   
...                               ...         ...             ...   
4995  Seven Brides for Seven Brothers           U         102 min   
4996                  Playing It Cool          15          94 min   
4997             A Moment to Remember     144 min  Drama, Romance   
4998                        Hairspray          PG          92 min   
4999             McCabe & Mrs. Miller           X         120 min   

                          genre  rating  \
0                         Drama     9.3   
1          Action, Crime, Drama     9.0   
2     Action, Adventure, Sci-Fi     8.8   
3                         Drama     8.8   
4                  Crime, Drama     8.9   
...                         ...     ...   
4995     Comedy, Drama, Musical     7.3   
4996            Comedy, Romance     5.9   
4997                        NaN     8.1   
4998      Comedy, Drama, Family     7.0   
4999             Drama, Western     7.6   

                                                  about           director  \
0     Two imprisoned men bond over a number of years...     Frank Darabont   
1     When the menace known as the Joker wreaks havo...  Christopher Nolan   
2     A thief who steals corporate secrets through t...  Christopher Nolan   
3     An insomniac office worker and a devil-may-car...      David Fincher   
4     The lives of two mob hitmen, a boxer, a gangst...  Quentin Tarantino   
...                                                 ...                ...   
4995  In 1850 Oregon, when a backwoodsman brings a w...      Stanley Donen   
4996  Unrequited love motivates a guy to write about...     Justin Reardon   
4997  A young couple's enduring love is tested when ...        John H. Lee   
4998  A 'pleasantly plump' teenager teaches 1962 Bal...        John Waters   
4999  A gambler and a prostitute become business par...      Robert Altman   

                                                  stars      votes gross_earn  
0     ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...  26,26,905    $28.34M  
1     ('Christian Bale',), ('Heath Ledger',), ('Aaro...  25,98,173   $534.86M  
2     ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...  23,04,062   $292.58M  
3     ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...  20,71,088    $37.03M  
4     ('John Travolta',), ('Uma Thurman',), ('Samuel...  20,11,013   $107.93M  
...                                                 ...        ...        ...  
4995  ('Jane Powell',), ('Howard Keel',), ('Jeff Ric...     25,112     $9.40M  
4996  ('Chris Evans',), ('Michelle Monaghan',), ('To...     28,515        NaN  
4997  ('Jung Woo-sung',), ('Son Ye-jin',), ('Jong-ha...     23,957        NaN  
4998  ('Sonny Bono',), ('Ruth Brown',), ('Divine',),...     21,536     $6.67M  
4999  ('Warren Beatty',), ('Julie Christie',), ('Ren...     25,097     $8.20M  

[5000 rows x 11 columns]

In [7]:
#clean data
# Filter to remove rows where 'certificate' contains runtime (values with 'min')
cleaned_movies[~cleaned_movies['certificate'].str.contains('min')]

poster  \
0     https://m.media-amazon.com/images/S/sash/4Fyxw...   
1     https://m.media-amazon.com/images/S/sash/4Fyxw...   
2     https://m.media-amazon.com/images/S/sash/4Fyxw...   
3     https://m.media-amazon.com/images/S/sash/4Fyxw...   
4     https://m.media-amazon.com/images/S/sash/4Fyxw...   
...                                                 ...   
4994  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4995  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4996  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4998  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4999  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                                title certificate  runtime  \
0            The Shawshank Redemption          15  142 min   
1                     The Dark Knight         12A  152 min   
2                           Inception         12A  148 min   
3                          Fight Club          18  139 min   
4                        Pulp Fiction          18  154 min   
...                               ...         ...      ...   
4994                            Heist          15   93 min   
4995  Seven Brides for Seven Brothers           U  102 min   
4996                  Playing It Cool          15   94 min   
4998                        Hairspray          PG   92 min   
4999             McCabe & Mrs. Miller           X  120 min   

                          genre  rating  \
0                         Drama     9.3   
1          Action, Crime, Drama     9.0   
2     Action, Adventure, Sci-Fi     8.8   
3                         Drama     8.8   
4                  Crime, Drama     8.9   
...                         ...     ...   
4994       Action, Crime, Drama     6.1   
4995     Comedy, Drama, Musical     7.3   
4996            Comedy, Romance     5.9   
4998      Comedy, Drama, Family     7.0   
4999             Drama, Western     7.6   

                                                  about           director  \
0     Two imprisoned men bond over a number of years...     Frank Darabont   
1     When the menace known as the Joker wreaks havo...  Christopher Nolan   
2     A thief who steals corporate secrets through t...  Christopher Nolan   
3     An insomniac office worker and a devil-may-car...      David Fincher   
4     The lives of two mob hitmen, a boxer, a gangst...  Quentin Tarantino   
...                                                 ...                ...   
4994  A father is without the means to pay for his d...         Scott Mann   
4995  In 1850 Oregon, when a backwoodsman brings a w...      Stanley Donen   
4996  Unrequited love motivates a guy to write about...     Justin Reardon   
4998  A 'pleasantly plump' teenager teaches 1962 Bal...        John Waters   
4999  A gambler and a prostitute become business par...      Robert Altman   

                                                  stars      votes gross_earn  
0     ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...  26,26,905    $28.34M  
1     ('Christian Bale',), ('Heath Ledger',), ('Aaro...  25,98,173   $534.86M  
2     ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...  23,04,062   $292.58M  
3     ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...  20,71,088    $37.03M  
4     ('John Travolta',), ('Uma Thurman',), ('Samuel...  20,11,013   $107.93M  
...                                                 ...        ...        ...  
4994  ('Robert De Niro',), ('Jeffrey Dean Morgan',),...     30,472        NaN  
4995  ('Jane Powell',), ('Howard Keel',), ('Jeff Ric...     25,112     $9.40M  
4996  ('Chris Evans',), ('Michelle Monaghan',), ('To...     28,515        NaN  
4998  ('Sonny Bono',), ('Ruth Brown',), ('Divine',),...     21,536     $6.67M  
4999  ('Warren Beatty',), ('Julie Christie',), ('Ren...     25,097     $8.20M  

[4970 rows x 11 columns]

In [8]:
cleaned_movies

poster  \
0     https://m.media-amazon.com/images/S/sash/4Fyxw...   
1     https://m.media-amazon.com/images/S/sash/4Fyxw...   
2     https://m.media-amazon.com/images/S/sash/4Fyxw...   
3     https://m.media-amazon.com/images/S/sash/4Fyxw...   
4     https://m.media-amazon.com/images/S/sash/4Fyxw...   
...                                                 ...   
4995  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4996  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4997  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4998  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4999  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                                title certificate         runtime  \
0            The Shawshank Redemption          15         142 min   
1                     The Dark Knight         12A         152 min   
2                           Inception         12A         148 min   
3                          Fight Club          18         139 min   
4                        Pulp Fiction          18         154 min   
...                               ...         ...             ...   
4995  Seven Brides for Seven Brothers           U         102 min   
4996                  Playing It Cool          15          94 min   
4997             A Moment to Remember     144 min  Drama, Romance   
4998                        Hairspray          PG          92 min   
4999             McCabe & Mrs. Miller           X         120 min   

                          genre  rating  \
0                         Drama     9.3   
1          Action, Crime, Drama     9.0   
2     Action, Adventure, Sci-Fi     8.8   
3                         Drama     8.8   
4                  Crime, Drama     8.9   
...                         ...     ...   
4995     Comedy, Drama, Musical     7.3   
4996            Comedy, Romance     5.9   
4997                        NaN     8.1   
4998      Comedy, Drama, Family     7.0   
4999             Drama, Western     7.6   

                                                  about           director  \
0     Two imprisoned men bond over a number of years...     Frank Darabont   
1     When the menace known as the Joker wreaks havo...  Christopher Nolan   
2     A thief who steals corporate secrets through t...  Christopher Nolan   
3     An insomniac office worker and a devil-may-car...      David Fincher   
4     The lives of two mob hitmen, a boxer, a gangst...  Quentin Tarantino   
...                                                 ...                ...   
4995  In 1850 Oregon, when a backwoodsman brings a w...      Stanley Donen   
4996  Unrequited love motivates a guy to write about...     Justin Reardon   
4997  A young couple's enduring love is tested when ...        John H. Lee   
4998  A 'pleasantly plump' teenager teaches 1962 Bal...        John Waters   
4999  A gambler and a prostitute become business par...      Robert Altman   

                                                  stars      votes gross_earn  
0     ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...  26,26,905    $28.34M  
1     ('Christian Bale',), ('Heath Ledger',), ('Aaro...  25,98,173   $534.86M  
2     ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...  23,04,062   $292.58M  
3     ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...  20,71,088    $37.03M  
4     ('John Travolta',), ('Uma Thurman',), ('Samuel...  20,11,013   $107.93M  
...                                                 ...        ...        ...  
4995  ('Jane Powell',), ('Howard Keel',), ('Jeff Ric...     25,112     $9.40M  
4996  ('Chris Evans',), ('Michelle Monaghan',), ('To...     28,515        NaN  
4997  ('Jung Woo-sung',), ('Son Ye-jin',), ('Jong-ha...     23,957        NaN  
4998  ('Sonny Bono',), ('Ruth Brown',), ('Divine',),...     21,536     $6.67M  
4999  ('Warren Beatty',), ('Julie Christie',), ('Ren...     25,097     $8.20M  

[5000 rows x 11 columns]

In [9]:
# Remove the dollar sign and "M"
cleaned_movies['gross_earn'] = cleaned_movies['gross_earn'].str.replace('$', '').str.replace('M', '')

# Convert the column to numeric
cleaned_movies['gross_earn'] = pd.to_numeric(cleaned_movies['gross_earn'])

In [10]:
print(cleaned_movies.shape)


(5000, 11)


In [11]:
# Clean the runtime column by removing non-numeric characters
cleaned_movies['runtime'] = cleaned_movies['runtime'].astype(str).str.replace(r'[^0-9]', '', regex=True)

# Convert the cleaned runtime column to numeric
cleaned_movies['runtime'] = pd.to_numeric(cleaned_movies['runtime'], errors='coerce')


In [12]:
print(cleaned_movies.shape)


(5000, 11)


In [13]:
cleaned_movies['votes'] = cleaned_movies['votes'].str.replace(',', '').astype(float)

In [14]:
print(cleaned_movies.shape)

(5000, 11)


In [15]:
cleaned_movies['star_count'] = cleaned_movies['stars'].apply(lambda x: len(eval(x)))

In [16]:
cleaned_movies['director_avg_earn'] = cleaned_movies.groupby('director')['gross_earn'].transform('mean')

In [17]:
print(cleaned_movies.isnull().sum())


poster                 0
title                  0
certificate            0
runtime               30
genre                 30
rating                 0
about                 14
director               0
stars                  0
votes                  0
gross_earn           415
star_count             0
director_avg_earn    204
dtype: int64


In [18]:
cleaned_movies.dtypes

poster                object
title                 object
certificate           object
runtime              float64
genre                 object
rating               float64
about                 object
director              object
stars                 object
votes                float64
gross_earn           float64
star_count             int64
director_avg_earn    float64
dtype: object

In [19]:
print(cleaned_movies.columns)

Index(['poster', 'title', 'certificate', 'runtime', 'genre', 'rating', 'about',
       'director', 'stars', 'votes', 'gross_earn', 'star_count',
       'director_avg_earn'],
      dtype='object')


In [20]:
cleaned_movies.head()

poster  \
0  https://m.media-amazon.com/images/S/sash/4Fyxw...   
1  https://m.media-amazon.com/images/S/sash/4Fyxw...   
2  https://m.media-amazon.com/images/S/sash/4Fyxw...   
3  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                      title certificate  runtime                      genre  \
0  The Shawshank Redemption          15    142.0                      Drama   
1           The Dark Knight         12A    152.0       Action, Crime, Drama   
2                 Inception         12A    148.0  Action, Adventure, Sci-Fi   
3                Fight Club          18    139.0                      Drama   
4              Pulp Fiction          18    154.0               Crime, Drama   

   rating                                              about  \
0     9.3  Two imprisoned men bond over a number of years...   
1     9.0  When the menace known as the Joker wreaks havo...   
2     8.8  A thief who steals corporate secrets through t...   
3     8.8  An insomniac office worker and a devil-may-car...   
4     8.9  The lives of two mob hitmen, a boxer, a gangst...   

            director                                              stars  \
0     Frank Darabont  ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...   
1  Christopher Nolan  ('Christian Bale',), ('Heath Ledger',), ('Aaro...   
2  Christopher Nolan  ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...   
3      David Fincher  ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...   
4  Quentin Tarantino  ('John Travolta',), ('Uma Thurman',), ('Samuel...   

       votes  gross_earn  star_count  director_avg_earn  
0  2626905.0       28.34           4          54.635000  
1  2598173.0      534.86           4         201.832222  
2  2304062.0      292.58           4         201.832222  
3  2071088.0       37.03           4          86.519000  
4  2011013.0      107.93           4          78.026250

In [21]:
#fill in missing values
cleaned_movies['runtime'] = cleaned_movies['runtime'].fillna(cleaned_movies['runtime'].median())

In [22]:

# fill in missing values with the mode (most common genre)
cleaned_movies['genre'] = cleaned_movies['genre'].fillna(cleaned_movies['genre'].mode()[0])

In [23]:
#replace missing about column data with no description avail.
cleaned_movies['about'] = cleaned_movies['about'].fillna('No description available')

In [24]:
#fill in missing gross earn with the median
cleaned_movies['gross_earn'] = cleaned_movies['gross_earn'].fillna(cleaned_movies['gross_earn'].median())

In [25]:
#fill in missing director_avg_earn with mean
cleaned_movies['director_avg_earn'] = cleaned_movies['director_avg_earn'].fillna(cleaned_movies['director_avg_earn'].mean())

In [26]:
print(cleaned_movies.isnull().sum())

poster               0
title                0
certificate          0
runtime              0
genre                0
rating               0
about                0
director             0
stars                0
votes                0
gross_earn           0
star_count           0
director_avg_earn    0
dtype: int64


In [ ]:
#save the cleaned dataset
cleaned_movies.to_csv('cleaned_movies.csv', index=False)

In [253]:
cleaned_movies.head()

poster  \
0  https://m.media-amazon.com/images/S/sash/4Fyxw...   
1  https://m.media-amazon.com/images/S/sash/4Fyxw...   
2  https://m.media-amazon.com/images/S/sash/4Fyxw...   
3  https://m.media-amazon.com/images/S/sash/4Fyxw...   
4  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                      title certificate  runtime                      genre  \
0  The Shawshank Redemption          15    142.0                      Drama   
1           The Dark Knight         12A    152.0       Action, Crime, Drama   
2                 Inception         12A    148.0  Action, Adventure, Sci-Fi   
3                Fight Club          18    139.0                      Drama   
4              Pulp Fiction          18    154.0               Crime, Drama   

   rating                                              about  \
0     9.3  Two imprisoned men bond over a number of years...   
1     9.0  When the menace known as the Joker wreaks havo...   
2     8.8  A thief who steals corporate secrets through t...   
3     8.8  An insomniac office worker and a devil-may-car...   
4     8.9  The lives of two mob hitmen, a boxer, a gangst...   

            director                                              stars  \
0     Frank Darabont  ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...   
1  Christopher Nolan  ('Christian Bale',), ('Heath Ledger',), ('Aaro...   
2  Christopher Nolan  ('Leonardo DiCaprio',), ('Joseph Gordon-Levitt...   
3      David Fincher  ('Brad Pitt',), ('Edward Norton',), ('Meat Loa...   
4  Quentin Tarantino  ('John Travolta',), ('Uma Thurman',), ('Samuel...   

       votes  gross_earn  star_count  director_avg_earn  
0  2626905.0       28.34           4          54.635000  
1  2598173.0      534.86           4         201.832222  
2  2304062.0      292.58           4         201.832222  
3  2071088.0       37.03           4          86.519000  
4  2011013.0      107.93           4          78.026250

In [55]:
# Drop unwanted columns
movies = movies.drop(['about', 'director', 'stars', 'poster', 'certificate', 'genre'], axis=1)

In [56]:
movies.head()

title  runtime  rating      votes  gross_earn  \
0  The Shawshank Redemption      142     9.3  2626905.0       28.34   
1           The Dark Knight      152     9.0  2598173.0      534.86   
2                 Inception      148     8.8  2304062.0      292.58   
3                Fight Club      139     8.8  2071088.0       37.03   
4              Pulp Fiction      154     8.9  2011013.0      107.93   

              genre_filtered  star_count  director_avg_earn  
0                      Drama           4          54.635000  
1       Action, Crime, Drama           4         201.832222  
2  Action, Adventure, Sci-Fi           4         201.832222  
3                      Drama           4          86.519000  
4                      Other           4          78.026250

In [57]:
movies = movies.drop(['title', 'genre_filtered'], axis=1)

In [60]:
movies.head()

runtime  rating      votes  gross_earn  star_count  director_avg_earn
0      142     9.3  2626905.0       28.34           4          54.635000
1      152     9.0  2598173.0      534.86           4         201.832222
2      148     8.8  2304062.0      292.58           4         201.832222
3      139     8.8  2071088.0       37.03           4          86.519000
4      154     8.9  2011013.0      107.93           4          78.026250

In [61]:
#load libraries
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import classification_report, confusion_matrix

In [63]:
threshold = movies['gross_earn'].median()  # Example threshold
movies['is_good'] = (movies['gross_earn'] > threshold).astype(int)
# Features (make sure these are the right features after preprocessing)
X = movies[['runtime', 'rating', 'votes', 'star_count', 'director_avg_earn', 'gross_earn', 'star_count']]  # features
y = movies['is_good']  # Target variable


In [64]:
#split the data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)

In [74]:
#feature scaling
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)


In [75]:
# Check the data types of the features
print(X_train.dtypes)


runtime                int64
rating               float64
votes                float64
star_count             int64
director_avg_earn    float64
gross_earn           float64
star_count             int64
dtype: object


In [80]:
# Check for missing values in the training and test sets
print(X_train.isnull().sum())
print(X_test.isnull().sum())


runtime              0
rating               0
votes                0
star_count           0
director_avg_earn    0
gross_earn           0
star_count           0
dtype: int64
runtime              0
rating               0
votes                0
star_count           0
director_avg_earn    0
gross_earn           0
star_count           0
dtype: int64


In [78]:
# Impute missing values for specific columns manually
X_train['director_avg_earn'] = X_train['director_avg_earn'].fillna(X_train['director_avg_earn'].mean())
X_train['gross_earn'] = X_train['gross_earn'].fillna(X_train['gross_earn'].mean())

X_test['director_avg_earn'] = X_test['director_avg_earn'].fillna(X_test['director_avg_earn'].mean())
X_test['gross_earn'] = X_test['gross_earn'].fillna(X_test['gross_earn'].mean())

# Check again for missing values
print(X_train.isnull().sum())
print(X_test.isnull().sum())


runtime              0
rating               0
votes                0
star_count           0
director_avg_earn    0
gross_earn           0
star_count           0
dtype: int64
runtime              0
rating               0
votes                0
star_count           0
director_avg_earn    0
gross_earn           0
star_count           0
dtype: int64


In [83]:
y_train

1446    0
2960    0
1882    0
627     0
1743    1
       ..
4630    0
1310    1
755     1
2042    0
3261    1
Name: is_good, Length: 3976, dtype: int32

In [81]:
from sklearn.impute import SimpleImputer

# Impute missing values with the mean for numerical columns
imputer = SimpleImputer(strategy='mean')

# Apply imputation to both training and test data
X_train_imputed = imputer.fit_transform(X_train)
X_test_imputed = imputer.transform(X_test)


In [84]:
#train the SVM model
# Initialize the SVM classifier
svm_model = SVC(kernel='linear', random_state=42)

# Fit the model to the training data
svm_model.fit(X_train_imputed, y_train)


SVC(kernel='linear', random_state=42)

In [86]:
y_pred = svm_model.predict(X_test_imputed)


In [87]:
# Print classification report
print(classification_report(y_test, y_pred))

# Print confusion matrix
print(confusion_matrix(y_test, y_pred))


              precision    recall  f1-score   support

           0       0.88      0.86      0.87       537
           1       0.84      0.86      0.85       457

    accuracy                           0.86       994
   macro avg       0.86      0.86      0.86       994
weighted avg       0.86      0.86      0.86       994

[[464  73]
 [ 62 395]]


Classification Report:
Precision: The proportion of true positive predictions out of all positive predictions made by the model.

For class 0: 88% of the predicted class 0 instances were actually 0.
For class 1: 84% of the predicted class 1 instances were actually 1.
Recall: The proportion of true positive predictions out of all actual positive instances.

For class 0: The model correctly identified 86% of all true 0 instances.
For class 1: The model correctly identified 86% of all true 1 instances.
F1-Score: The harmonic mean of precision and recall, providing a single score to evaluate the balance between precision and recall.

For class 0: The F1-score is 0.87.
For class 1: The F1-score is 0.85.
Support: The number of occurrences of each class in the dataset.

Class 0: 537 instances.
Class 1: 457 instances.
Accuracy: The overall correctness of the model, which is 86% in this case.

Confusion Matrix:
464: True positives for class 0 (correctly predicted as 0).
395: True positives for class 1 (correctly predicted as 1).
73: False positives for class 0 (predicted 0 but actually 1).
62: False negatives for class 1 (predicted 1 but actually 0).
Insights:
The model is doing well with an accuracy of 86%, but there are still some misclassifications, particularly between class 0 and class 1.
The precision and recall scores are quite close for both classes, suggesting that the model is balanced in terms of its predictions for both classes.